src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1657,10 +1659,48 @@ void Syscollector::push(const std::string& data)
         auto rawData{data};
         Utils::replaceFirst(rawData, "dbsync ", "");
         const auto buff{reinterpret_cast<const uint8_t*>(rawData.c_str())};
+        bool pushMessage = true;
+        std::string component;
+        std::string topic;
+        time_t id;
+        bool parseSuccesfull = true;
 
         try
         {
-            m_spRsync->pushMessage(std::vector<uint8_t> {buff, buff + rawData.size()});
+            component = Utils::splitIndex(rawData, ' ', 0);
+            topic = Utils::splitIndex(rawData, ' ', 1);
+            const nlohmann::json syncInfoJson = nlohmann::json::parse(Utils::splitIndex(rawData, ' ', 2));
+            id = std::stol(syncInfoJson.at("id").dump());
+        }
+        catch (...)
+        {
+            parseSuccesfull = false;
+        }
+
+        try
+        {
+            if (parseSuccesfull)
+            {
+                const auto itComponent = m_lastSyncMsg.find(component);
+
+                if (itComponent == m_lastSyncMsg.end())
```

Comment:
We can reduce the scope of the iterator
```suggestion
                if (const auto it = m_lastSyncMsg.find(component); it == m_lastSyncMsg.end(); it == m_lastSyncMsg.end())
```

But also, given that this is a map, we can use count instead
```suggestion
                if (m_lastSyncMsg.count(component) == 0)
```

Reply:
I'll try